web/templates/web/study-detail.html

```diff
@@ -31,23 +30,85 @@
     <script>
         function childSelected(selected) {
             var participateButton = document.getElementById('participate-button');
-            selected.value === 'none' ? participateButton.disabled = true : participateButton.disabled = false;
+            if (selected.value === 'none') {
+                participateButton.disabled = true;
+                document.getElementById('too-old').style.display = 'none';
+                document.getElementById('too-young').style.display = 'none';
+            } else {
+                participateButton.disabled = false;
+            }
             participateButton.value = selected.value;
+            modifyTemplateWithAgeCheck(selected);
+        }
+
+        function modifyTemplateWithAgeCheck(selected) {
+            document.getElementById('too-old').style.display = 'none';
+            document.getElementById('too-young').style.display = 'none';
+            var birthday = selected.value.split('|')[1];
+            var age = calculateAgeInDays(birthday);
+            if (!isEligible(age)) {
+                if (isOldEnough(age)) {
+                    document.getElementById('too-old').style.display = 'inline-block';
+                } else {
+                    document.getElementById('too-young').style.display = 'inline-block';
+                }
+            }
         }
 
         function portToParticipate(selected) {
-            window.location.href = "/studies/" + "{{ study.uuid }}" + '/' + selected.value;
+            window.location.href = "/studies/" + "{{ study.uuid }}" + '/' + selected.value.split('|')[0];
```

Comment:
I moved building the url into a javascript function because the "Participate" button needed to be conditionally disabled and I don't think you can disable links, just buttons.